lib/index.js

```diff
@@ -407,13 +405,28 @@ api.getAll = async ({
   let result = get(record, typeKey, []);
 
   result = Array.isArray(result) ? result : [result];
+  const allTokens = result;
+  let tokens = [];
+  const expiredTokens = [];
   const now = new Date();
-  if(filterExpiredTokens) {
-    result = result.filter(token => {
-      return token.expires > now;
-    });
+  for(const token of result) {
+    if(token.expires > now) {
+      if(filterExpiredTokens) {
+        tokens.push(token);
+      }
+    } else {
+      expiredTokens.push(token);
+    }
```

Comment:
So looking at this more, I think continuing to have the `filterExpiredTokens` flag is likely to confuse devs.

If we're going to always run this loop that filters the output and then always return `allTokens`, `tokens`, and `expiredTokens` -- what's the point of the flag? Depending on how you set it the value of just `tokens` will change? I expect that to be confusing. Can we remove the `filterExpiredTokens` flag and adjust whatever else was calling this before to just use the appropriate value out of `result`?

For example, if something was setting `filterExpiredTokens: false` then it should just use `allTokens`, it should not make `allTokens` and `tokens` have the same value via a flag and then use `tokens`. If something was setting `filterExpiredTokens: true`, then it should just use `tokens` if all it needs are the unexpired ones.

Reply:
Well we don't currently have anything setting `filterExpiredTokens` to `false` any longer. So if there is not a use case currently for that then I think removing the flag makes sense. However, if there is a use case for that, we are still going to have some sort of flag that specifies whether to use "filter" the tokens or not right? I think it's more confusing to move this out of the getAll function. I think getAll needs to be doing to filtering as that makes the most sense to me.

Reply:
@JoshDunnDev, I'm not suggesting we move the filtering out, I'm saying that we don't need the flag. We should just make it so that the return value is *always*:

```js
{
  allTokens, // always all the tokens, not filtered
  tokens, // always all the unexpired tokens, does not change based on a flag
  expiredTokens, // always all the expired tokens
}
```

Reply:
Alright I have gone ahead and made the change to remove the flag.